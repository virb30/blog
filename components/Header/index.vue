<template>
	<header class="h-20 border-0 border-b border-solid border-gray-400 dark:border-gray-100">
		<div class="max-w-7xl h-20 my-0 mx-auto p-4 flex items-center justify-between">
			<Logo class="max-w-xs hidden md:block lg:block xl:block" />
			<nav class="md:ml-20 h-20">
				<NuxtLink
					exact
					to="/"
					active-class="active"
					class="hover:text-black dark-hover:text-white inline-block relative py-0 px-3 h-20 leading-20 transition-color-2 mx-1"
				>
					Home
				</NuxtLink>
				<NuxtLink
					to="/about"
					active-class="active"
					class="hover:text-black dark-hover:text-white inline-block relative py-0 px-3 h-20 leading-20 transition-color-2 mx-1"
				>
					Sobre
				</NuxtLink>
				<NuxtLink
					to="/blog"
					active-class="active"
					class="hover:text-black dark-hover:text-white inline-block relative py-0 px-3 h-20 leading-20 transition-color-2 mx-1"
				>
					Blog
				</NuxtLink>
			</nav>
			<div class="ml-auto p-4">
				<SwitchMode />
			</div>
		</div>
	</header>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
});
</script>

<style src="./styles.module.css" scoped></style>
