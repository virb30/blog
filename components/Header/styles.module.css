.leading-20 {
	line-height: 5rem;
}

.transition-color-2 {
	transition: color 0.2s;
}

nav > a:hover,
nav > a.active {
	@apply font-bold;
}

.active::after {
	content: "";
	@apply w-full absolute left-0 bg-indigo-500 rounded-t rounded-b-none h-1;
	bottom: 1px;
}
