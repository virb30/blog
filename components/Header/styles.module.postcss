.headerContainer {
	@apply h-20 border-0 border-b border-gray-800 border-solid;
}

.headerContent {
	@apply max-w-6xl h-20 my-0 mx-auto p-8 flex items-center;

	& nav {
		@apply ml-20 h-20;

		& a {
			@apply inline-block relative py-0 px-2 h-20 text-gray-300;

			line-height: 5rem;
			transition: color 0.2s;

			& + a {
				@apply ml-8;
			}

			&:hover {
				@apply text-white;
			}

			&.active {
				@apply text-white font-bold;
			}

			&.active::after {
				content: "";
				@apply w-full absolute left-0 bg-indigo-500;
				height: 3px;
				border-radius: 3px 3px 0 0;
				bottom: 1px;
			}
		}
	}

	& button {
		@apply ml-auto;
	}
}
