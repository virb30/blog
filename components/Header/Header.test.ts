import { mount, RouterLinkStub } from '@vue/test-utils';
import { describe, expect, it } from 'vitest';
import Header from './index.vue';

describe('Header Component', () => {
    it('should render childs', () => {
        const wrapper = mount(Header, {
            global: {
                stubs: {
                    ThemeSwitcher: true,
                    Logo: true,
                    NuxtLink: RouterLinkStub,
                    'font-awesome-icon': true
                },
            }
        });

        const links = wrapper.findAllComponents(RouterLinkStub);

        expect(wrapper.html()).toMatchSnapshot();
        expect(links.length).toEqual(3);
    });
});
