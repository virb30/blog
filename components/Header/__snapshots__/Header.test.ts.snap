// Vitest Snapshot v1

exports[`Header Component > should render childs 1`] = `
"<header class=\\"h-20 border-0 border-b border-solid border-gray-400 dark:border-gray-100\\" data-v-f744be1a=\\"\\">
  <div class=\\"max-w-7xl h-20 my-0 mx-auto p-4 flex items-center justify-between\\" data-v-f744be1a=\\"\\">
    <logo-stub class=\\"max-w-xs hidden md:block lg:block xl:block\\" data-v-f744be1a=\\"\\"></logo-stub>
    <nav class=\\"md:ml-20 h-20\\" data-v-f744be1a=\\"\\"><a exact=\\"\\" active-class=\\"active\\" class=\\"hover:text-black dark:hover:text-white inline-block relative py-0 px-3 h-20 leading-20 transition-color-2 mx-1\\" data-v-f744be1a=\\"\\"> Home </a><a active-class=\\"active\\" class=\\"hover:text-black dark:hover:text-white inline-block relative py-0 px-3 h-20 leading-20 transition-color-2 mx-1\\" data-v-f744be1a=\\"\\"> Sobre </a><a active-class=\\"active\\" class=\\"hover:text-black dark:hover:text-white inline-block relative py-0 px-3 h-20 leading-20 transition-color-2 mx-1\\" data-v-f744be1a=\\"\\"> Blog </a></nav>
    <div class=\\"ml-auto p-4\\" data-v-f744be1a=\\"\\">
      <theme-switcher-stub color-mode=\\"dark\\" data-v-f744be1a=\\"\\"></theme-switcher-stub>
    </div>
  </div>
</header>"
`;
